### PR TITLE
Droth 527 busstop interface use viite instead of vkm to locate oth stops in road registry

### PR DESCRIPTION
**Pull request by MAC-CGI**
Droth 527 busstop interface use viite instead of vkm to locate oth stops in road registry

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/GeometryTransform.scala [hunk 1]
```diff
@@ -56,275 +46,65 @@ object RoadSide {
...
-    val params = Map(VkmRoad -> road, VkmRoadPart -> roadPart, VkmDistance -> distance, VkmTrackCode -> track.map(_.value),
-    "x" -> Option(coord.x), "y" -> Option(coord.y), VkmSearchRadius -> Some(searchDistance.getOrElse(DefaultToleranceMeters)),
-      VkmRoadNumberInterval -> roadNumberInterval(includePedestrian, road))
-    request(vkmUrl + urlParams(params)) match {
-      case Left(address) => mapFields(address)
-      case Right(error) => throw new VKMClientException(error.toString)
+    val qfilter = (road, includePedestrian) match {
+      case (Some(road), Some(false)) => "AND road_number = " + road
+      case (Some(road), Some(true)) => "AND road_number BETWEEN " + NonPedestrianRoadNumbersMin + " AND " + NonPedestrianRoadNumbersMax
+      case (None, Some(true)) => "AND road_number BETWEEN " + AllRoadNumbersMin + " AND " + AllRoadNumbersMax
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/GeometryTransform.scala]: We belive this two conditions can be removed
```
case (Some(road), Some(true)) => "AND road_number BETWEEN " + NonPedestrianRoadNumbersMin + " AND " + NonPedestrianRoadNumbersMax
case (None, Some(true)) => "AND road_number BETWEEN " + AllRoadNumbersMin + " AND " + AllRoadNumbersMax
```

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/GeometryTransform.scala [hunk 2]
```diff
@@ -56,275 +46,65 @@ object RoadSide {
...
-        "Missing mandatory field in response: %s".format(
-          fieldName))
-    }
-    try {
-      value.get.toString.toInt
-    } catch {
-      case e: NumberFormatException =>
-        throw new VKMClientException("Invalid value in response: %s, Int expected, got '%s'".format(fieldName, value.get))
+    val address = roadAddress match {
+      case Some(addr) if (addr.track.value == assetSideCode || addr.track == Track.Combined) => RoadAddress(Some(municipalityCode.toString), addr.roadNumber.toInt, addr.roadPartNumber.toInt, addr.track, (addr.startAddrMValue + (mValue - addr.startMValue)).toInt, None)
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/GeometryTransform.scala]: Assuming that two road address can not be equal, with different trac code to the same link id. I belive that the following condition can be removed.
```
if (addr.track.value == assetSideCode || addr.track == Track.Combined)
```

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/GeometryTransform.scala [hunk 3]
```diff
@@ -56,275 +46,65 @@ object RoadSide {
...
+      case Some(addr) if (addr.track.eq(Track.Unknown)) => throw new RoadAddressException ("Invalid value for Track: %d".format( addr.track.value))
+      case None  => throw new RoadAddressException("Invalid value for Track")
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/GeometryTransform.scala]: The exception message can be changed to "No road address found".

**Comment by joanacoelho** [comment]
Seems ok to me

**Comment by marquesfm** [comment]
ok to me

**Pull request merged**